impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.okta.sdk.impl.oauth2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.okta.commons.http.HttpException;
+import com.okta.commons.http.HttpMethod;
+import com.okta.commons.lang.Assert;
+import com.okta.commons.lang.Strings;
+import com.okta.sdk.impl.config.ClientConfiguration;
+import com.okta.sdk.oauth2.ErrorResponse;
+import com.okta.sdk.oauth2.OAuth2AccessToken;
+import okhttp3.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+
+import static com.okta.sdk.impl.util.OAuth2Utils.createSignedJWT;
+
+public class AccessTokenRetrieverServiceImpl implements AccessTokenRetrieverService {
+    private static final Logger log = LoggerFactory.getLogger(AccessTokenRetrieverServiceImpl.class);
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private ClientConfiguration clientConfiguration;
+
+    public AccessTokenRetrieverServiceImpl(ClientConfiguration clientConfiguration) {
+        Assert.notNull(clientConfiguration);
+        this.clientConfiguration = clientConfiguration;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OAuth2AccessToken getOAuth2AccessToken()
+        throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+        String signedJwt = createSignedJWT(clientConfiguration);
+        String scope = String.join(" ", clientConfiguration.getScopes());
+
+        MediaType mediaType = MediaType.parse("application/x-www-form-urlencoded");
+        RequestBody body = RequestBody.create(mediaType, "");
+
+        OkHttpClient client = new OkHttpClient().newBuilder().build();
```

Comment:
Use the existing Http framework within the code base. Don't add another HTTP client framework (this logic is already abstract out, and will automatically perform retry requests)

Reply:
I ran into issues using the existing http client framework trying to POST to the token endpoint with an empty body. Spent a lot of time trying to debug that and eventually resorted to this easy way :)

Reply:
Chat with me about the problems you are having, this whole project is based around making HTTP requests.  Using a different HTTP client introduces a few problems:
* different timeout configuration
* no retry support (or limited in the case of okhttp)
* adds a dependency on a project where we have an alternative solution already on the classpath

Reply:
I've used:

`<dependency>
            <groupId>com.okta.commons</groupId>
            <artifactId>okta-http-okhttp</artifactId>
        </dependency>`

Reply:
This is fixed.